apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -158,25 +156,83 @@ impl RawResponse {
                 debug_request,
                 data,
             } => {
-                let error = graphql::Error::builder()
-                    .message("Request failed".to_string())
-                    .extension_code("CONNECTOR_FETCH")
-                    .extension("service", connector.id.subgraph_name.clone())
-                    .extension(
-                        "http",
-                        Value::Object(Map::from_iter([(
-                            "status".into(),
-                            Value::Number(parts.status.as_u16().into()),
-                        )])),
-                    )
-                    .extension(
-                        "connector",
-                        Value::Object(Map::from_iter([(
-                            "coordinate".into(),
-                            Value::String(connector.id.coordinate().into()),
-                        )])),
-                    )
-                    .path::<Path>((&key).into())
+                let mut message = "Request failed".to_string();
+                let mut extensions = None;
+
+                // Do we have errors settings for this connector?
+                let inputs = key.inputs().merge(
+                    &connector.response_variables,
+                    &connector.response_headers,
+                    connector.config.as_ref(),
+                    context,
+                    Some(parts.status.as_u16()),
+                    supergraph_request,
+                    Some(&parts),
+                );
+
+                // Do we have a error message mapping set for this connector?
+                if let Some(message_selection) = &connector.error_settings.message {
+                    // TODO: In the future, we'll want to add to the debug context. However, we'll need a "v2" debug payload before we can do that.
+                    let (res, _apply_to_errors) = message_selection.apply_with_vars(&data, &inputs);
+
+                    message = res
+                        .unwrap_or_else(|| Value::Null)
+                        .as_str()
+                        .unwrap_or_default()
+                        .to_string();
+                }
```

Comment:
```suggestion
                let message = if let Some(message_selection) = &connector.error_settings.message {
                    // TODO: In the future, we'll want to add to the debug context. However, we'll need a "v2" debug payload before we can do that.
                    let (res, _apply_to_errors) = message_selection.apply_with_vars(&data, &inputs);

                    res
                        .unwrap_or_else(|| Value::Null)
                        .as_str()
                        .unwrap_or_default()
                        .to_string()
                } else {
                  "Request failed".to_string()
                }
```

Just to not allocate the default if we don't need it.